docs/src/config_directories.md

```diff
@@ -2,6 +2,19 @@
 
 This section allows overriding some directory paths.
 
+## `cache_dir`
+
+Override the cache directory. Remember to use an absolute path. Variable
+expansion will not be performed on the path. If the directory does not yet
+exist, it will be created.
```

Comment:
Don't you think it's still worthwhile to document that this is the expected behavior? For me, it would indicate that this behavior is by design, not by concidence. (If you still think the sentence should be removed, I can do that.)

Reply:
Actually, I'd keep the sentence. Intuitively, as a user, I would expect the cache directory to be created when downloading the cache, or that an error message would be returned if it doesn't yet exist. However, it's created as soon as tealdeer is run:

![screenshot-20220924-221721](https://user-images.githubusercontent.com/105168/192117206-2c297f2e-fbbc-4e94-afe2-40af056af112.png)

Reply:
Hm, I don't know what motivated my initial comment, probably felt like it was more "feature-bragging" than useful, but you are right that it makes sense to have documentation to point to if questions arise.

However, I think that your screenshot is a misbehavior, we shouldn't create it unless we need to, or? Let's fix that before the next release (if not in this PR :D)

Reply:
The advantage of the current solution is that we never have to check for existence of the cache directory before we use it, we initialize it in the beginning and then assume it's there. This simlifies some code paths.

However, after second thought I agree that this might not be ideal from a user perspective. Additionally, we always need to expect races on the file system. I changed it to create the cache directory lazily.